cmake/platform/windows.cmake

```diff
@@ -88,3 +88,28 @@ if(MSVC)
   # https://blogs.msdn.microsoft.com/vcblog/2018/04/09/msvc-now-correctly-reports-__cplusplus/
   add_compile_options(/Zc:__cplusplus)
 endif()
+
+#
+# Helper macros added to the catkin system to replace Windows lack of shebang support.
+#
+# Code courtesy of Yujin Robot.  Derived from https://github.com/ros-windows/win_ros
+#
+
+# distutils does not have support for 'entry_points'
+macro(add_windows_python_batch_helper name)
+  set(PYTHON_TARGET ${name})
+  configure_file(${catkin_EXTRAS_DIR}/templates/python_wrapper.bat.in
+                 ${CATKIN_DEVEL_PREFIX}/bin/${name}.bat)
+  install(PROGRAMS ${CATKIN_DEVEL_PREFIX}/bin/${PYTHON_TARGET}.bat 
+          DESTINATION ${CATKIN_GLOBAL_BIN_DESTINATION})
+endmacro()
+
+# some python scripts need an executable, not a batch file
```

Comment:
Please write a docblock similar to the ones in other CMake functions, e.g. https://github.com/ros/catkin/blob/0627d42c63e20d0bca3ba38c777acece30616924/cmake/catkin_package.cmake#L1-L82 (it doesn't have to be that verbose).